internal/builders/istio.go

```diff
@@ -2,323 +2,327 @@ package builders
 	value *v1beta1.AuthorizationPolicy
 }
 
-func (aps *AuthorizationPolicySpec) Get() *v1beta1.AuthorizationPolicy {
+func (aps *AuthorizationPolicySpecBuilder) Get() *v1beta1.AuthorizationPolicy {
 	return aps.value
 }
 
-func (aps *AuthorizationPolicySpec) From(val *v1beta1.AuthorizationPolicy) *AuthorizationPolicySpec {
+func (aps *AuthorizationPolicySpecBuilder) FromAP(val *v1beta1.AuthorizationPolicy) *AuthorizationPolicySpecBuilder {
 	aps.value = val
 	return aps
 }
 
-func (aps *AuthorizationPolicySpec) Selector(val *Selector) *AuthorizationPolicySpec {
-	aps.value.Selector = val.Get()
+func (aps *AuthorizationPolicySpecBuilder) WithSelector(val *apiv1beta1.WorkloadSelector) *AuthorizationPolicySpecBuilder {
+	aps.value.Selector = val
 	return aps
 }
 
-func (aps *AuthorizationPolicySpec) Rule(val *Rule) *AuthorizationPolicySpec {
-	aps.value.Rules = append(aps.value.Rules, val.Get())
+func (aps *AuthorizationPolicySpecBuilder) WithRule(val *v1beta1.Rule) *AuthorizationPolicySpecBuilder {
+	aps.value.Rules = append(aps.value.Rules, val)
 	return aps
 }
 
-// RuleBuilder returns builder for istio.io/api/security/v1beta1/Rule type
-func RuleBuilder() *Rule {
-	return &Rule{
+// NewRuleBuilder returns builder for istio.io/api/security/v1beta1/Rule type
+func NewRuleBuilder() *RuleBuilder {
+	return &RuleBuilder{
 		value: &v1beta1.Rule{},
 	}
 }
 
-type Rule struct {
+type RuleBuilder struct {
 	value *v1beta1.Rule
 }
 
-func (r *Rule) Get() *v1beta1.Rule {
+func (r *RuleBuilder) Get() *v1beta1.Rule {
 	return r.value
 }
 
-func (r *Rule) RuleFrom(val *RuleFrom) *Rule {
-	r.value.From = append(r.value.From, val.Get())
+func (r *RuleBuilder) WithFrom(val *v1beta1.Rule_From) *RuleBuilder {
+	r.value.From = append(r.value.From, val)
 	return r
 }
 
-func (r *Rule) RuleTo(val *RuleTo) *Rule {
-	r.value.To = append(r.value.To, val.Get())
+func (r *RuleBuilder) WithTo(val *v1beta1.Rule_To) *RuleBuilder {
+	r.value.To = append(r.value.To, val)
 	return r
 }
 
-func (r *Rule) RuleCondition(val *RuleCondition) *Rule {
-	r.value.When = *val.Get()
+func (r *RuleBuilder) WithWhenCondition(val *v1beta1.Condition) *RuleBuilder {
+	r.value.When = append(r.value.When, val)
 	return r
 }
 
-// RuleFromBuilder returns builder for istio.io/api/security/v1beta1/Rule_From type
-func RuleFromBuilder() *RuleFrom {
-	return &RuleFrom{
+// NewFromBuilder returns builder for istio.io/api/security/v1beta1/Rule_From type
+func NewFromBuilder() *FromBuilder {
+	return &FromBuilder{
 		value: &v1beta1.Rule_From{},
 	}
 }
 
-type RuleFrom struct {
+type FromBuilder struct {
 	value *v1beta1.Rule_From
 }
 
-func (rf *RuleFrom) Get() *v1beta1.Rule_From {
+func (rf *FromBuilder) Get() *v1beta1.Rule_From {
 	return rf.value
 }
 
-func (rf *RuleFrom) Source() *RuleFrom {
+// WithJWTRequirement adds RequestPrincipals = "*" requirement, forcing requests to use JWT authorization
+func (rf *FromBuilder) WithJWTRequirement() *FromBuilder {
```

Comment:
I would still go with leaving the From builder, since having it (at least for me) better visualises how our code translates to IstioAP. Not having a builder for one of the three supported keywords ("To", "When", "From") might not be clear